### DISCUSSION TITLE
Multiple contributors on a page

### DISCUSSION THREAD

**Discussion by brutaldigital** [thegetty/quire#767] [General]
Dear Erin / Greg,

I'm hoping you could indicate the best means by which to assign contributors to a section of text on a page. I'm looking to replicate a structure similar such as this in Quire, where texts are contributed by named individuals:  https://britishartstudies.ac.uk/issues/issue-index/issue-1/conversation

The `name-title-block` format described at https://quire.getty.edu/docs-v1/contributors/ is closest to the output I'm looking for, but how would you suggest the MD files be structured? All contributors listed on a single page, or individual pages for each contribution with a custom template to pull them together? Or something else?

Thanks,
Tom

**Comment by Erin-Cecele** [comment]
@brutaldigital Thank you for posting this question, Tom. We will be in touch very soon!

**Comment by geealbers** [comment]
I would list all the contributors in the `publication.yaml` file with an `id` and then create a simple shortcode or include to reference them in individual page sections as needed. This would fit with Quire's existing data structure, and would let you more easily customize the markup to have whatever structure you like. The existing contributors shortcode is really designed around looking up contributors by broad types rather than being able to pick and choose them more granularly. You'd want to also be sure to list all the contributors of a given page by their `id` in the page's frontmatter YAML to ensure they're added to the page metadata for search engines since the new shortcode/include wouldn't do that for you otherwise.

Here's a quick example using a [Liquid](https://shopify.github.io/liquid/) include. Though you could also do this in JavaScript as noted below.

In `publication.yaml`:

```yaml
contributor:
  - id: jblume
    first_name: Judy
    last_name: Blume
  - id: bcleary
    first_name: Beverly
    last_name: Cleary
```

A custom includes file `_includes/byline.liquid` where you can grab any of the contributor attributes you need and output them in any HTML markup you'd like. Here I just have them as a list of names:

```liquid
{%- assign byline_ids = id | split: ", " -%}
<ul>
{%- for id in byline_ids -%}
  {%- for contributor in publication.contributor -%}
    {%- if contributor.id == id -%}
    <li>{{ contributor.first_name }} {{ contributor.last_name }}</li>
    {%- endif -%}
{%- endfor -%}
{%- endfor -%}
</ul>
```

And then this is the liquid code you can use in any Markdown file, with a list of comma-separated contributor `id`s that match ones in the `publication.yaml` file, to output the contributor list:

```liquid
{% include "byline" id: "jblume, bcleary" %}
```

Or you could also do all this in JavaScript, in which case the code would go in `_plugins/shortcodes/byline.js` and be registered in `_plugins/shortcodes/index.js` with the others. And the shortcode would be `{% byline "jblume, bcleary" %}`.
